lib/iris/cube.py

```diff
@@ -755,6 +751,10 @@ def __init__(self, data, standard_name=None, long_name=None,
         # Cell Measures
         self._cell_measures_and_dims = []
 
+        self.fill_value = fill_value
+
+        self._dtype = dtype
```

Comment:
Do you have a reference for this ?
I had not heard of this as a "rule" --  I can see why it might be important in other languages, but why in Python ?

As Python is so dynamic, I can also see there ceratainly _could_ be good reasons for breaking this in particular cases, where one property may need to exist so that we can leverage generic code to set another state aspect.
For instance, here in cube init, we might require dtype and/or shape to be set up so that we could use ```cube.data = data```.
(Which actually would make good sense now, since data assignment has become generic to real or lazy data).